WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewViewModel.swift

```diff
@@ -187,22 +253,81 @@ final class ProductDetailPreviewViewModel: ObservableObject {
+        guard let generatedAIProduct else {
+            return
+        }
+
+        guard let option = options[safe: selectedOptionIndex] else {
+            return
+        }
+
+        switch field {
+        case .name:
+            options[selectedOptionIndex] = NameSummaryDescOption(name: generatedAIProduct.names[selectedOptionIndex],
+                                                                 shortDescription: option.shortDescription,
+                                                                 description: option.description)
+        case .shortDescription:
+
+            options[selectedOptionIndex] = NameSummaryDescOption(name: option.name,
+                                                                 shortDescription: generatedAIProduct.shortDescriptions[selectedOptionIndex],
+                                                                 description: option.description)
+        case .description:
+            options[selectedOptionIndex] = NameSummaryDescOption(name: option.name,
+                                                                 shortDescription: option.shortDescription,
+                                                                 description: generatedAIProduct.descriptions[selectedOptionIndex])
+        }
+    }
+
+    private func saveCurrentOption() {
+        options[selectedOptionIndex] = NameSummaryDescOption(name: productName,
+                                                             shortDescription: productShortDescription,
+                                                             description: productDescription)
+    }
+}
+
 // MARK: - Product details for preview
 //
 private extension ProductDetailPreviewViewModel {
-    func observeGeneratedProduct() {
-        generatedProductSubscription = $generatedProduct
-            .compactMap { $0 }
-            .sink { [weak self] product in
-                guard let self else { return }
-                self.updateProductDetails(with: product)
+    func observeSelectedOption() {
+        $selectedOptionIndex
+            .combineLatest($options)
+            .sink { [weak self] index, options in
+                guard let self,
+                      let option = options[safe: index] else {
+                    return
+                }
+
+                self.productName = option.name
+                self.productDescription = option.description
+                self.productShortDescription = option.shortDescription
             }
+            .store(in: &subscriptions)
     }
 
-    func updateProductDetails(with product: Product) {
-        productName = product.name
-        productShortDescription = product.shortDescription ?? ""
-        productDescription = product.fullDescription ?? ""
+    func displayAIProductDetails(aiProduct: AIProduct) throws {
+        guard
+            aiProduct.names.count > 0,
+            aiProduct.shortDescriptions.count > 0,
+            aiProduct.descriptions.count > 0 else {
```

Comment:
Nit: We can use `isNotEmpty` for brevity here.

Reply:
Done in f4567a75133aedb8883e110eb145fc070a9e61d5